src/settings-ui/Settings.UI/ViewModels/CmdNotFoundViewModel.cs

```diff
@@ -96,28 +147,98 @@ public void RunPowerShellScript(string powershellArguments)
             }
 
             CommandOutputLog = outputLog;
+            return outputLog;
         }
 
-        public void CheckPowershellVersion()
+        public void CheckCommandNotFoundRequirements()
         {
-            var arguments = $"-NoProfile -NonInteractive -Command $PSVersionTable";
-            RunPowerShellScript(arguments);
+            var ps1File = AssemblyDirectory + "\\Assets\\Settings\\Scripts\\CheckCmdNotFoundRequirements.ps1";
+            var arguments = $"-NoProfile -ExecutionPolicy Unrestricted -File \"{ps1File}\"";
+            var result = RunPowerShellScript("pwsh.exe", arguments);
+
+            if (result.Contains("PowerShell 7.4 or greater detected."))
+            {
+                IsPowerShell7Detected = true;
+            }
+            else if (result.Contains("PowerShell 7.4 or greater not detected."))
+            {
+                IsPowerShell7Detected = false;
+            }
+            else if (result.Contains("pwsh.exe"))
+            {
+                // Likely an error saying there was an error starting pwsh.exe, so we can assume Powershell 7 was not detected.
+                CommandOutputLog += "PowerShell 7.4 or greater not detected. Installation instructions can be found on https://learn.microsoft.com/powershell/scripting/install/installing-powershell-on-windows \r\n";
+                IsPowerShell7Detected = false;
+            }
+
+            if (result.Contains("WinGet Client module detected."))
+            {
+                IsWinGetClientModuleDetected = true;
+            }
+            else if (result.Contains("WinGet Client module not detected."))
+            {
+                IsWinGetClientModuleDetected = false;
+            }
+
+            if (result.Contains("Command Not Found module is registered in the profile file."))
+            {
+                IsCommandNotFoundModuleInstalled = true;
+            }
+            else if (result.Contains("Command Not Found module is not registered in the profile file."))
+            {
+                IsCommandNotFoundModuleInstalled = false;
+            }
+        }
+
+        public void InstallPowerShell7()
+        {
+            var arguments = $"-NoProfile -Command \"winget install --id Microsoft.Powershell --source winget\"";
+            var result = RunPowerShellScript("powershell.exe", arguments);
```

Comment:
Always english on every os language setting?

Reply:
if you're talking about succesfully installed, then yes. This is output we show from our ps script

Reply:
Actually for PowerShell 7, it isn't 🤔